storage/inmem/inmem.go

```diff
@@ -121,20 +121,19 @@ func (db *store) Truncate(ctx context.Context, txn storage.Transaction, _ storag
 					return err
 				}
 
-				_, err := underlying.Read(update.Path[:len(update.Path)-1])
+				err = underlying.Write(storage.AddOp, update.Path, obj)
```

Comment:
Hmm this reads funny. If I understand this correctly, we're attempting the write, and if the path isn't there, we makedir it, and write again. Is MakeDir costly? We could always do it first and write afterwards 🤔

Reply:
So the idea is that the first `Write` call will fail as the storage won't exist but the next ones should work w/o needing to read the path first.

Reply:
I'm trying an alternate approach where we don't rely on the error status from the Write call. Let's see how that goes. Thanks for the review!